### PR TITLE
Do not create truncate triggers on foreign tables

### PR DESCRIPTION
**Pull request by agedemenli**
Backporting https://github.com/citusdata/citus/pull/6100

### CODE REVIEW

File: src/backend/distributed/sql/udfs/citus_finish_pg_upgrade/11.0-1.sql [hunk 1]
```diff
@@ -104,7 +104,7 @@ BEGIN
     --
     -- register triggers
     --
-    FOR table_name IN SELECT logicalrelid FROM pg_catalog.pg_dist_partition
+    FOR table_name IN SELECT logicalrelid FROM pg_catalog.pg_dist_partition JOIN pg_class ON (logicalrelid = oid) WHERE relkind <> 'f'
```
Comment (onderkalaci) [inline src/backend/distributed/sql/udfs/citus_finish_pg_upgrade/11.0-1.sql:107]: If we do this, how are already Citus 11 users would get this?

We should need something like: https://github.com/citusdata/citus/pull/6037/files

@hanefi do you agree?

Comment (hanefi) [reply src/backend/distributed/sql/udfs/citus_finish_pg_upgrade/11.0-1.sql]: Yes, that is correct. I will start reviewing and testing if the updated migration versions are fine now.

File: src/backend/distributed/sql/downgrades/citus--11.0-4--11.0-3.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+#include "../udfs/citus_finish_pg_upgrade/11.0-3.sql"
```
Comment (hanefi) [inline src/backend/distributed/sql/downgrades/citus--11.0-4--11.0-3.sql:1]:
```
/home/circleci/project/src/backend/distributed/sql/downgrades/citus--11.0-4--11.0-3.sql:1:10: fatal error: ../udfs/citus_finish_pg_upgrade/11.0-3.sql: No such file or directory
    1 | #include "../udfs/citus_finish_pg_upgrade/11.0-3.sql"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This needs an update here.

```suggestion
#include "../udfs/citus_finish_pg_upgrade/11.0-1.sql"
```

**Review by hanefi — APPROVED** [review]

**Review by onderkalaci — APPROVED** [review]

**Comment by onderkalaci** [comment]
Given that we support PG14 on Citus 10.2, I guess we should need another PR for that one as well? @agedemenli  what do you think?

**Comment by agedemenli** [comment]
> Given that we support PG14 on Citus 10.2, I guess we should need another PR for that one as well? @agedemenli what do you think?

Makes sense. I'll do that

**Comment by hanefi** [comment]
> > Given that we support PG14 on Citus 10.2, I guess we should need another PR for that one as well? @agedemenli what do you think?
> 
> Makes sense. I'll do that

Note that it will be slightly more tricky as you will need to introduce a new migration version for `10.2`, and you will need to update both `main` and `release-11.1` branches so that there are migration paths from that newly introduced migration version.

You might as well consider holding off on #6100 until you backport the fix to 10.2 as well. What do you think?

**Pull request merged**